### PR TITLE
fix: Custom network chain id not converted to hexadecimal format

### PR DESCRIPTION
**Pull request by tommasini**
## **Description**
This [PR](https://github.com/MetaMask/metamask-mobile/pull/8234/files#diff-6db807cd50e056502de0cd9a1277d4a284458f3eee610d6ed1a5115da9a8cdb4) was merged some days before the hexadecimal chain id format, and those changes were missed when the hexadecimal chain id PR was merged.

This PR addresses the conversion to hexadecimal format and also a non handled promise when nickname is undefined when a custom network is added.

## **Related issues**

Fixes: NA

## **Manual testing steps**

1. Go to add custom network
2. Enter fields needed ([like Gnosis network info](https://chainlist.org/?search=gnosis))
3. The user should see a bottom sheet to confirm the intention of adding that network
4. The user should see a bottom sheet to switch to that network
5. User should be switched to that network if confirmed it

## **Screenshots/Recordings**

https://recordit.co/ECgXE9ZHke

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [ ] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/components/Views/Settings/NetworksSettings/NetworkSettings/index.js [hunk 1]
```diff
@@ -884,13 +884,11 @@ class NetworkSettings extends PureComponent {
...
 
     const url = new URL(rpcUrl);
-    const decimalChainId = this.getDecimalChainId(chainId);
 
     const selectedNetwork = {
       rpcUrl: url.href,
-      chainId: decimalChainId,
       ticker,
-      nickname,
+      nickname: nickname ?? '',
```
Comment (NicolasMassart) [inline app/components/Views/Settings/NetworksSettings/NetworkSettings/index.js:891]: Q; is this related?

Comment (tommasini) [reply app/components/Views/Settings/NetworksSettings/NetworkSettings/index.js]: Is related, On this PR I'm addressing all the changes missed on this [PR](https://github.com/MetaMask/metamask-mobile/pull/8234/files#diff-6db807cd50e056502de0cd9a1277d4a284458f3eee610d6ed1a5115da9a8cdb4) and when nickname was undefined was not being handled correctly, but instead of this fallback, it's better than we just check on the networkChecker if the nickname is defined!

https://recordit.co/QWj7TbyIHX

pushed that changes !

**Review by NicolasMassart — DISMISSED** [review]
Looks good to me.

**Review by NicolasMassart — APPROVED** [review]
